R/retrieve_patient_id.R

```diff
@@ -0,0 +1,165 @@
+                                 output_table_name = NULL) {
+  conn <- connect_oracle() # Connect to database
+
+  # Check table format (ORACLE table)
+  if (is.character(ben_table_name) & dbExistsTable(conn, ben_table_name)) {
+    # pass
+  } else {
+    stop("ben_table_name must be a character name of an existing ORACLE table")
+  }
+
+  # Check table contents and retrieve BEN_IDT_ANO in both archived and non-archived tables
+  if ("BEN_IDT_ANO" %in% (tbl(conn, ben_table_name) %>% colnames())) {
+    idt <- tbl(conn, ben_table_name) %>%
+      select(BEN_IDT_ANO) %>%
+      distinct()
+  } else if ("BEN_NIR_PSA" %in% (tbl(conn, ben_table_name) %>% colnames())) {
+    # Consider BEN_RNG_GEM for join if available
+    join_cond <- ifelse(
+      all(c("BEN_NIR_PSA", "BEN_RNG_GEM") %in% (tbl(conn, ben_table_name) %>%
+        colnames())),
+      c("BEN_NIR_PSA", "BEN_RNG_GEM"),
+      c("BEN_NIR_PSA")
+    )
+    psa_reg <- tbl(conn, ben_table_name) %>%
+      select(BEN_NIR_PSA) %>%
+      left_join(tbl(conn, "IR_BEN_R"), by = join_cond) %>%
+      select(BEN_IDT_ANO) %>%
+      distinct()
+  # Consider archived table if requested
+  if (check_arc_table) {
+    psa_arc <- tbl(conn, ben_table_name) %>%
+      select(BEN_NIR_PSA) %>%
+      left_join(tbl(conn, "IR_BEN_R_ARC"), by = join_cond) %>%
+      select(BEN_IDT_ANO) %>%
+      distinct()
+
+    idt <- union_all(psa_reg, psa_arc) %>%
+      distinct()
+  } else { # Only consider non-archived table
+		idt <- psa_reg, psa_arc %>%
+			distinct()
+  }
+  } else {
+    stop("ben_table_name must contain (at least) either BEN_IDT_ANO or BEN_NIR_PSA")
+  }
+
+  # Retrieve corresponding BEN_NIR_PSA
+  idt_psa_reg <- idt %>%
+    inner_join(tbl(conn, "IR_BEN_R"), by = "BEN_IDT_ANO") %>%
+    filter(!is.na(BEN_NIR_PSA))
+  # Consider archived table if requested
+ if (check_arc_table) {
+   idt_psa_arc <- idt %>%
+     inner_join(tbl(conn, "IR_BEN_R_ARC"), by = "BEN_IDT_ANO") %>%
+     filter(!is.na(BEN_NIR_PSA))
+   idt_psa <-  union_all(idt_psa_reg, idt_psa_arc)
+ } else { # Only consider non-archived table
+   idt_psa <- idt_psa_reg
+ }
+  # Assessment of exclusion criteria
+  idt_psa <- idt_psa %>%
+    distinct() %>%
+    collect() %>%
+    group_by(BEN_NIR_PSA) %>%
+    mutate(psa_w_multiple_idt = n_distinct(BEN_IDT_ANO) > 1) %>%
+    group_by(BEN_IDT_ANO) %>%
+    mutate(
+      psa_w_multiple_idt = any(psa_w_multiple_idt), # flag IDT with PSA associated w/ multiple IDT
+      cdi_nir_00 = !is.na(BEN_CDI_NIR) & BEN_CDI_NIR == "00", # flag non-fictive NIR
+      nir_ano_defined = !is.na(BEN_NIR_ANO), # flag defined BEN_NIR_ANO
+      birth_date_variation = (n_distinct(BEN_NAI_ANN) > 1 || n_distinct(BEN_NAI_MOI) > 1), # flag IDT with multiple birth dates
+      sex_variation = n_distinct(BEN_SEX_COD) > 1 # flag IDT with multiple sex codes
+    )
+  # Return tibble
+  if (is.null(output_table_name)) {
+    return(idt_psa)
+    dbDisconnect(conn)
+  } else { # Save table in Oracle
```

Comment:
Tu pourrais rajouter un message utilisateur ici comme on l'a fait dans d'autres fonctions : 

`message(
      glue::glue("Results saved to table {output_table_name} in Oracle.")
    )`